out_writeapi_test.go

```diff
@@ -38,3 +56,136 @@ func TestFLBPluginRegister(t *testing.T) {
+	return m.GetWriteStreamFunc(ctx, req, opts...)
+}
+
+func (m *MockManagedWriterClient) Close() error {
+	return m.client.Close()
+}
+
+func (m *MockManagedWriterClient) BatchCommitWriteStreams(ctx context.Context, req *storagepb.BatchCommitWriteStreamsRequest, opts ...gax.CallOption) (*storagepb.BatchCommitWriteStreamsResponse, error) {
+	return m.client.BatchCommitWriteStreams(ctx, req, opts...)
+}
+
+func (m *MockManagedWriterClient) CreateWriteStream(ctx context.Context, req *storagepb.CreateWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+	return m.client.CreateWriteStream(ctx, req, opts...)
+}
+
+// TestFLBPluginInit tests the FLBPluginInit function
+func TestFLBPluginInit(t *testing.T) {
+	var currChecks OptionChecks
+	mockClient := &MockManagedWriterClient{
+		NewManagedStreamFunc: func(ctx context.Context, opts ...managedwriter.WriterOption) (*managedwriter.ManagedStream, error) {
+			currChecks.calledNewManagedStream++
+			if len(opts) == 6 {
+				currChecks.numInputs = true
+			}
+			return nil, nil
+
+		},
+		GetWriteStreamFunc: func(ctx context.Context, req *storagepb.GetWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+			currChecks.calledGetWriteStream++
+			return &storagepb.WriteStream{
+				Name: "mockstream",
+				TableSchema: &storagepb.TableSchema{
+					Fields: []*storagepb.TableFieldSchema{
+						{Name: "Time", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+						{Name: "Text", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+					},
+				},
+			}, nil
+		},
+	}
+
+	originalFunc := getClient
+	getClient = func(ctx context.Context, projectID string) (ManagedWriterClient, error) {
+		currChecks.calledGetClient++
+		return mockClient, nil
+	}
+	defer func() { getClient = originalFunc }()
+
+	patch1 := monkey.Patch(output.FLBPluginConfigKey, func(plugin unsafe.Pointer, key string) string {
+		log.Println("Mock out.FLBPluginConfigKey called")
+		switch key {
+		case "ProjectID":
+			currChecks.configProjectID = true
+			return "DummyProjectId"
+		case "DatasetID":
+			currChecks.configDatasetID = true
+			return "DummyDatasetId"
+		case "TableID":
+			currChecks.configTableID = true
+			return "DummyTableId"
+		case "Max_Chunk_Size":
+			currChecks.configMaxChunkSize = true
+			return "0"
+		case "Max_Queue_Requests":
+			currChecks.configMaxQueueRequests = true
+			return "0"
+		case "Max_Queue_Bytes":
+			currChecks.configMaxQueueSize = true
+			return "0"
+		default:
+			return ""
+		}
+	})
+	defer patch1.Unpatch()
+
+	patch2 := monkey.Patch(output.FLBPluginSetContext, func(plugin unsafe.Pointer, ctx interface{}) {
+		currChecks.calledSetContext++
+	})
+	defer patch2.Unpatch()
+
+	plugin := unsafe.Pointer(nil)
+	initsize := len(configMap)
```

Comment:
It feels odd to be accessing an internal member variable used by the plugin. Can you add a test-only method on the plugin that provides the instance count and use that instead?

Reply:
Done.